### PR TITLE
[PS Move] cellGem updates

### PR DESCRIPTION
**Pull request by Megamouse**
This may or may not have an impact on any PSMove game.

It fixes some calibration screens. For example "Fast Draw Showdown" and "Carnival Island"

### CODE REVIEW

**Comment by legend800** [comment]
House of the Dead Overkill has been the one PS Move game that doesn't work with the current implementation (can't get past cali screen). 

With this PR, it changes behavior a little. Before it seemed to skip the first screen which prompts Select button, which is 1 of 3 unmapped buttons currently with rpcs3 (Square, Select, PS Home button). With PR, it keeps you there, which might be an improvement but not sure.

PR:
2021-07-30 19_43_04-Window

Master:
2021-07-30 19_48_23-PS3

The problem looks to be the in-game cursor doesn't budge when the mouse cursor does (always centered):
2021-07-30 19_50_01-FPS_ 30 00 _ Vulkan _ 0 0 17-12558 _ The House Of The Dead_ OVERKILL™ Extended C

**Comment by Megamouse** [comment]
You can press select by pressing the "back" thumb button on your mouse with this PR. After calibration (which sometimes takes a couple of tries) you can then proceed as usual. The game can still only be played with a controller though

**Comment by legend800** [comment]
Brilliant, confirmed Back mouse button maps to Select. What about Square (last one) - maybe Forward mouse button? Is the rest of this still accurate:
o	Left click = T
o	Right click = ~ button
o	Middle click = X 
o	Left click + Middle click = Start
o	Right click + Middle click = Triangle
o	Right click + Middle click + Left click = Circle

For HOTD Overkill, yeah cursor just won't move and get stuck here until you move to gamepad:
2021-07-31 00_31_51-FPS_ 30 01 _ Vulkan _ 0 0 17-local_bu-local_build _ The House Of The Dead_ OVERK

**Comment by Megamouse** [comment]
I can add it tomorrow or so

**Comment by Ralf-Rollan** [comment]
Megamouse any chance that you can help to work on PS Eye Camera to work on RPCS3? I love playing “Wonderbook: Book of Potions/Spells” but it requires the camera so you can move the actual book, it works along the PS Move

**Comment by Megamouse** [comment]
Not anytime soon

**Comment by Megamouse** [comment]
Well I added the square button and made it work a bit different.
It should now be like this: (which is probably gonna be a good or bad experience, depending on the game)

Leftclick: T-Button
Rightclick: Move button
Middleclick:  X Button
Thumb-Back: Select
Thumb-Forward: Start
"Mouse Button 6" OR "Leftclick+Rightclick": Circle
"Mouse Button 7" OR "Leftclick+Middleclick": Square
"Mouse Button 8" OR "Middleclick+Rightclick": Triangle

**Comment by legend800** [comment]
Actually Deadstorm Pirate's Calibration screen is broken.  Can't press Move ~ button to start, Circle doesn't work also like input didn't get initiated properly at this point, so you're soft locked here now. 
2021-08-01 19_29_29-FPS_ 59 76 _ Vulkan _ 0 0 17-local_bu-local_build _ TIME CRISIS_ RAZING STORM  B

Works perfect on today's Master build.

**Comment by Megamouse** [comment]
I've switched the square and cirlce buttons with the start and select buttons. this seems to have less conflicts during gameplay from what i can tell.

**Comment by legend800** [comment]
Deadstorm Pirates cali screen is still broken, but TC Razing and 4 are fine on the latest build. 

Verified the new button layout which is improved:
o	Left click: T
o	Right click: ~/Move button
o	Middle click: X 
o	Back: Circle
o	Forward: Square
o	Left + Right click (or button 6): Select
o	Left + Middle click (or 7): Start
o	Middle + Right click (or 8): Triangle

I would suggest one more tweak - swap Triangle and Square (or maybe with X). I don't have anything ps move game that uses Square but at least Time Crisis Razing, 4, Deadstorm Pirates uses Triangle.

File: rpcs3/Emu/Cell/Modules/cellGem.cpp [hunk 1]
```diff
@@ -1089,17 +1223,25 @@ error_code cellGemPrepareVideoConvert(vm::cptr<CellGemVideoConvertAttribute> vc_
...
 	{
-		return CELL_GEM_ERROR_INVALID_ALIGNMENT;
+		if (!vc.video_data_out || ((vc.conversion_flags & CELL_GEM_COMBINE_PREVIOUS_INPUT_FRAME) && !vc.buffer_memory))
+		{
+			return CELL_GEM_ERROR_INVALID_PARAMETER;
+		}
 	}
 
+	// TODO:
+	//if (vc.video_data_out & 0x1f || vc.buffer_memory & 0xff)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellGem.cpp:1240]: It seems that alignment values were 1 bit off (should be 0xf and 0x7f respectively)

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellGem.cpp:1240]: Can you provide the code please?

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellGem.cpp:1240]: It's a vm::ptr<void>

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellGem.cpp:1240]: .aligned(x)

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellGem.cpp:1240]: yes. but is it aligned(0xf) and aligned(0xff)

Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellGem.cpp:1240]: `aligned(16)` and `aligned(128)`, these are common alignment values (vector/cache line)

**Comment by legend800** [comment]
Why was this merged if it broke a game? As mentioned above this PR breaks the cali screen on Deadstorm Pirates causing a soft lock now. Perfect on Master.

2021-08-04 12_16_52-Commits · RPCS3_rpcs3 — Mozilla Firefox

**Comment by Megamouse** [comment]
Because it should fix more games than this one.
And it was clearly wrong before, so there was no point in waiting.

I'll try to find out why this one isn't working anymore, but I feel like it's not directly caused by the stuff I added and rather caused by something that wasn't implemented yet anyway.

**Comment by dudesh34** [comment]
Hi mouse, thanks for your great work:
1. do you think you can add 2 psmoves setting so we can play games with 2 players like time crisis 4?

**Comment by Megamouse** [comment]
No. Not unless there's proper psmove support. But that's not implemented.

**Comment by dudesh34** [comment]
so...whats the situation now, is dead storm still broken? what was the last version the all the razing storm games works perfect?  i will download it, thanks.

**Comment by legend800** [comment]
Yeah it got fixed right after this. Just use the latest builds.

**Comment by dudesh34** [comment]
> Yeah it got fixed right after this. Just use the latest builds.

thanks a lot!

**Comment by dudesh34** [comment]
> Yeah it got fixed right after this. Just use the latest builds.

1.can you please paste your ahk for the new layout?
2.does it fixed the "Shake" problem in house of the dead 4?

**Comment by legend800** [comment]
The controls are pasted above. This does not add a simulated shake. Please continue conversation on discord or the forums..

**Comment by dudesh34** [comment]
for me "deadstorm pirate" is still broken on latest rpcs3 version... no calibration screen and mouse not respondung.

**Comment by legend800** [comment]
Works perfect on today's build.



•	Controls:
o	Left click: T
o	Right click: ~/Move button
o	Middle click: X 
o	Back: Circle
o	Forward: Square
o	Left + Right click (or mouse button 6): Select
o	Left + Middle click (or 7): Start
o	Middle + Right click (or 8): Triangle

**Pull request merged**